uint8 type
string source
uint32 data_length
string data
uint8 priority
---
bool success